### PR TITLE
Samples: Edge-to-edge issue on API 35

### PR DESCRIPTION
**Pull request by shubham7109**
## Description

> [Edge-to-edge is enforced](https://developer.android.com/about/versions/15/behavior-changes-15#edge-to-edge) on Android 15 (API level 35) and higher once your app targets SDK 35. If your app is not already edge-to-edge, portions of your app may be obscured and you must handle insets. Depending on the app, this work may or may not be significant.

This PR is to help avoid enforcing edge-to-edge on samples which currently do not support it, mainly XML samples. This is done using a v35 styles with `windowOptOutEdgeToEdgeEnforcement` set to true. To keep changes to a minimal, this opt-out rule will apply to the entire application (For XML & Compose samples). See screenshots for visual changes: 

|                       | XML                                                                                           | XML                                                                                            | Compose                                                                                           |
|-----------------------|-----------------------------------------------------------------------------------------------|------------------------------------------------------------------------------------------------|---------------------------------------------------------------------------------------------------|
| edge to edge enabled  | XML_false | XML2_false | COMPOSE_false |
| edge to edge disabled | XML_true  | XML2_true  | COMPOSE_true  |

Notes: 
- XML samples extend all the way to the status bar when edge-to-edge is enabled, this would be undesired behavior with the toolbar overlapping the Developer License text.   
- The bottom navigation bar overlaps with UI in XML samples.   
- Compose samples remain relatively unaffected, this is due to all samples using a `Scaffold` which manages edge to edge with the TopAppBar. 
- This PR changes it so that edge-to-edge is enforced only on the sample viewer but not on the samples themselves. 

## Links and Data
Issue: [Link](https://devtopia.esri.com/runtime/kotlin/issues/4917)

## Changes made: 
- Added a `res/values-v35/styles.xml` file to the samples-lib which includes the opt-out rule.
- Updated the `AppTheme` to be added on the root application manifest, which applies the theme to all activities/samples. 
- Updated `AppTheme` to use a `BaseAppTheme` and v35 style uses its respective theme. 
- Removed duplicate `sampleviewer/ui/theme/Theme.kt` and replaced with `sampleslib.theme.SampleAppTheme`
- Fixed a minor issue with screenshots not loading due to `v.next` branch change. 

## What To Review
- Run the sample viewer using a API 35 emulator/device to visualize the affected changes.
- Does the solution to "opt-out" seem reasonable? If enabled, we would need to redesign XML samples around the system and navigational bars.

### CODE REVIEW

**Review by TADraeseke — COMMENTED** [review]
Not spotting any edge obfuscation issues. I did spot an issue in the progress bar

And in that same screenshot there's an odd thing happening at the top of the screen too.

File: samples/generate-geodatabase-replica-from-feature-service/src/main/res/layout/generate_geodatabase_dialog_layout.xml [hunk 1]
```diff
@@ -17,7 +17,6 @@
             android:layout_marginEnd="8dp"
             android:layout_marginBottom="8dp"
             android:indeterminate="false"
-            android:theme="@style/AppTheme"
```
[the remark this answers is not in the document]
Comment (shubham7109) [reply samples/generate-geodatabase-replica-from-feature-service/src/main/res/layout/generate_geodatabase_dialog_layout.xml:20]: We do not have a shared progress dialog for XML samples, only a reusable Composable dialog.

File: samples/generate-offline-map-using-android-jetpack-workmanager/src/main/res/layout/offline_job_progress_dialog_layout.xml [hunk 2]
```diff
@@ -16,7 +16,6 @@
             android:layout_marginEnd="8dp"
             android:layout_marginBottom="8dp"
             android:indeterminate="false"
-            android:theme="@style/AppTheme"
```
[the remark this answers is not in the document]
Comment (shubham7109) [reply samples/generate-offline-map-using-android-jetpack-workmanager/src/main/res/layout/offline_job_progress_dialog_layout.xml:19]: What specifically are you noticing with this sample/dialog? 

Generate offline map uses the shared composable dialog component, in which I found a UI issue with the progress bar. Related PR Fix: https://github.com/Esri/arcgis-maps-sdk-kotlin-samples/pull/257#discussion_r1833220234

[the remark this answers is not in the document]
Comment (shubham7109) [reply samples/generate-offline-map-using-android-jetpack-workmanager/src/main/res/layout/offline_job_progress_dialog_layout.xml:19]: That seems to be part of [M3 design](https://m3.material.io/components/progress-indicators/overview), they call it the "stop indicator" 😅

[the remark this answers is not in the document]
Comment (TADraeseke) [reply samples/generate-offline-map-using-android-jetpack-workmanager/src/main/res/layout/offline_job_progress_dialog_layout.xml:19]: Haha wow ok!

**Review by TADraeseke — APPROVED** [review]
LGTM!

File: app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/model/Sample.kt [hunk 3]
```diff
@@ -109,7 +109,7 @@ data class Sample(
             // Otherwise, function should be modified to return list of URLs for each image
             val modifiedJsonSampleName = sampleName.replace(" ", "-").lowercase()
             val imageFileName = imageArray.first().toString().replace("\"", "")
-            return "https://raw.githubusercontent.com/Esri/arcgis-maps-sdk-kotlin-samples/v.next/$modifiedJsonSampleName/$imageFileName"
+            return "https://raw.githubusercontent.com/Esri/arcgis-maps-sdk-kotlin-samples/v.next/samples/$modifiedJsonSampleName/$imageFileName"
```
Comment (sorenoid) [inline app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/model/Sample.kt:112]: was this an intentional change?

**Review by sorenoid — APPROVED** [review]
LGTM, one sanity check for you

File: app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/model/Sample.kt (hunk 3 above)
Comment (shubham7109) [reply app/src/main/java/com/esri/arcgismaps/kotlin/sampleviewer/model/Sample.kt:112]: @sorenoid Yes, this change allows us to render screenshots in the sample viewer by grabbing the image from the `v.next` branch which now lives in `samples/<samples-name>/<file-name>`

> Fixed a minor issue with screenshots not loading due to v.next branch change.

**Pull request merged**